scripts/generate_upload_sbom.sh

```diff
@@ -95,16 +93,31 @@ echo "Platforms:" "${platforms[@]}"
 echo "S3 Path: $s3_path"
 
 for platform in "${platforms[@]}"; do
-  s3_path_platform_dependent="$s3_path/${platform////_}"
-  file_name="${image_name}_${tag_name}_${platform////_}.json"
+  os=${platform%/*}
+  arch=${platform#*/}
+
+  s3_path_platform_dependent="$s3_path/${os}_${arch}"
+  file_name="${image_name}_${tag_name}_${os}_${arch}.json"
+  
+  digest=$(docker manifest inspect $image_pull_spec | jq '.manifests[] | select(.platform.architecture == "'$arch'" and .platform.os == "'$os'")' | jq -r .digest)
+
+  echo "Verifying image signature before generating SBOM for $image_pull_spec ($platform)"
+  IMG=$image_pull_spec@$digest SIGNATURE_REPO=$repo_name/$image_name ${SCRIPT_DIR}/verify.sh
+
   echo "Generating SBOM for $image_pull_spec ($platform) and uploading to $s3_path_platform_dependent"
 
   if generate_sbom "$image_pull_spec" "$platform" "$file_name"; then
+    echo "Done generating SBOM for $image_pull_spec ($platform)"
```

Comment:
we probably want to tweak the file target location to be the one expected at release time:
`docs/releases/v${VERSION}/${os}-${arch}.sbom.json`